command/connect/envoy/exec_test.go

```diff
@@ -193,6 +186,11 @@ func TestHelperProcess(t *testing.T) {
 
 		limitProcessLifetime(2 * time.Minute)
 
+		// This is another level of gross - we are relying on `consul` being on path
+		// and being the correct version but in general that is true under `make
+		// test`. We already make the same assumption for API package tests.
+		testSelfExecOverride = "consul"
```

Comment:
I have thought about this numerous times too. Couldn't we during testing look in ./bin within the source directory for the binary instead of assuming the one on the path is the correct one.

Thats probabably a future task and not something to overhaul with this PR though.